### PR TITLE
Added NPC kill counts with message display

### PR DESCRIPTION
**Pull request by ipkpjersi**
This is a very cool system and allows for future expandability too, I have a couple of ideas building on this later on like an interface for boss NPC kills and an interface for slayer NPC kills.

### CODE REVIEW

**Comment by ipkpjersi** [comment]
For the record, out of all my PRs I opened, this is probably my favorite one.

P.S. Kill messages might pair nicely with my screenshots feature.

File: 2006Scape Server/src/main/java/com/rs2/game/players/Player.java [hunk 1]
```diff
@@ -144,6 +144,24 @@ public abstract class Player {
 	private GateHandler gateHandler = new GateHandler();
 	private SingleGates singleGates = new SingleGates();
 	private DoubleGates doubleGates = new DoubleGates();
+	
+	private Map<Integer, Integer> npcKillCounts = new HashMap<>();
+	public boolean displayBossKcMessages = true;
```
Comment (Dark98) [inline 2006Scape Server/src/main/java/com/rs2/game/players/Player.java:149]: this should be false by default

Comment (ipkpjersi) [reply 2006Scape Server/src/main/java/com/rs2/game/players/Player.java:149]: In that case it should be a client setting and a flag.

Comment (ipkpjersi) [reply 2006Scape Server/src/main/java/com/rs2/game/players/Player.java:149]: Oh wait it's the server nevermind me I'm tired lol

Comment (ipkpjersi) [reply 2006Scape Server/src/main/java/com/rs2/game/players/Player.java:149]: I already made it a flag, cool, I'll disable it by default.

**Pull request merged**